frontend/components/product-list/sections/RelatedPostsSection/index.tsx

```diff
@@ -55,37 +53,20 @@ export function RelatedPostsSection({ tags = [] }: RelatedPostsSectionProps) {
    );
 }
 
-function usePosts(tags: string[]) {
-   const previousTags = usePrevious(tags);
-
-   const [state, setState] = useSafeSetState<{
-      posts: Post[];
-      isLoading: boolean;
-      error?: string;
-   }>({
-      posts: [],
-      isLoading: true,
-   });
+const computeRelatedPostsKey = (tags: string[]) => [
+   `related_posts?data=${encodeURIComponent(JSON.stringify({ tags }))}`,
+];
 
-   React.useEffect(() => {
-      if (!isEqual(tags, previousTags)) {
-         fetchPosts(tags)
-            .then((posts) => {
-               setState({
-                  posts,
-                  isLoading: false,
-                  error: undefined,
-               });
-            })
-            .catch((error) => {
-               setState((current) => ({
-                  ...current,
-                  isLoading: false,
-                  error: error.message,
-               }));
-            });
+function usePosts(tags: string[]) {
+   const query = useQuery(
+      computeRelatedPostsKey(tags),
```

Comment:
Looks like react-query supports [`array-keys`](https://react-query-v3.tanstack.com/guides/query-keys#array-keys) which we should probably use instead of making a url-like string. We can sort the tags array so the ky is deterministic.